src/python/grpcio_testing/grpc_testing/__init__.py

```diff
@@ -0,0 +1,441 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Fixtures for use in testing gRPC Python-using application code."""
+
+import abc
+import collections
+
+import grpc
+import six
+
+
+class ChannelRpc(six.with_metaclass(abc.ABCMeta)):
+    """Fixture for an in-progress RPC invoked by a system under test.
+
+    Enables users to "play a server" for such RPCs by reading requests,
+    supplying responses, and supplying status.
+    """
+
+    @abc.abstractmethod
+    def invocation_metadata(self):
+        """Accesses the metadata passed by the system under test.
+
+        Returns:
+          The invocation metadata supplied by the system under test at
+            RPC-invocation-time.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def initial_metadata(self, initial_metadata):
+        """Accepts the initial metadata to be passed to the system under test.
+
+        Args:
+          initial_metadata: The RPC's initial metadata to be passed to the
+            system under test.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def take_request_as_message(self):
+        """Draws one of the requests added to the RPC by the system under test.
+
+        Successive calls to this method return requests in the same order in
+        which the system under test added them to the RPC.
+
+        Returns:
+          A deserialized request message added to the RPC by the system under
+            test.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def add_responses_as_messages(self, responses):
```

Comment:
How about `XYZ()` and `XYZ_as_bytes()` instead of `XYZ_as_messages()` and `XYZ_as_bytes()`?. I think by default `request` implies `as_messages` and is cleaner, leaving room for a separate `_as_bytes`.